cmake/r.cmake

```diff
@@ -0,0 +1,44 @@
+#################################################
+# (R) R.h Rinternals.h Rembedded.h Rexts.h etc. #
+#################################################
+option(BUILD_WITH_R "R Enabled" ${BUILD_WITH_ALL})
+message("BUILD_WITH_R: ${BUILD_WITH_R}")
+
+if(BUILD_WITH_R AND NOT N3DS AND NOT BAREMETALAPI)
+  set(TIC_BUILD_WITH_R TRUE)
+
+  if(NOT BUILD_STATIC)
+    include("cmake/renv.cmake")
+
+    set(R_LIB_DIR "/usr/include/R")
```

Comment:
Instead of hard-coding the paths, it's better to run the particular R you would like to link with and ask it about its include path, something like `Rscript -e 'cat(R.home("include"))'`.

Reply:
Yes, the paths I determined with `R CMD config --cppflags` and `R CMD config --ldflags`. I had much difficulty with CMake, despite reading some of the manual and beginning the tutorial until I felt comfortable to try again (after linking against a library in one of the introductory tutorial steps).

I have written an additional CMake file to generate a header which I can include to set the environment variables appropriately during runtime when they aren't set otherwise. At the moment I want to focus on getting a working integration running on my own system before I worry further about making it portable; it was a struggle to get where I am now.

I do know about the need for environment variables, libraries, etc. that R is dependent upon for its operation. I'll focus on them again later.

Reply:
> Yes, the paths I determined with `R CMD config --cppflags` and `R CMD config --ldflags`.

By this I mean that I'm aware of the commands and will use them later. For now, hard-coding was a troubleshooting methodology to simply ensure that they're there.